### DISCUSSION TITLE
Add two-step verification inside the chatbot

### DISCUSSION THREAD

**Discussion by AzizDXT** [open-webui/open-webui#7642] [General]
What is the best way to add two-step verification inside the chatbot, such that for example the student sends his username, then sends a verification code to his email via the API, then writes the verification code, and then receives the student’s data. What is the best way to do this process? Because I have faced many difficulties and so far I have not been able to do anything at all.

**Comment by unicorn667** [comment]
use the login page so you can use any sso provider with mfa
https://docs.openwebui.com/features/sso/
